### PR TITLE
Add basic celery-rabbitmq structure

### PR DESCRIPTION
**Pull request by vivekanand1101**
Signed-off-by: Vivek Anand <vivekanand1101@gmail.com>

### CODE REVIEW

File: pyconpune/tasks.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import os
+from celery import Celery
+
+import logging
+_log = logging.getLogger(__name__)
+
+
+if os.environ.get('CELERY_BROKER_URL'):
```
Comment (sayanchowdhury) [inline pyconpune/tasks.py]: Can we push this to the settings file?

File: pyconpune/tasks.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from celery import Celery
+
+import logging
+
+import settings
```
Comment (sayanchowdhury) [inline pyconpune/tasks.py:5]: You need to import from django.conf rather than just importing settings

File: pyconpune/tasks.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+else:
+    broker_url = 'amqp://localhost:5672'
+
+app = Celery('tasks', broker=broker_url)
+
+
+@app.task()
+def send_mail():
+    ''' Send the emails in an async manner '''
+    print 'Sending mails bro'
```
Comment (sayanchowdhury) [inline pyconpune/tasks.py:21]: This method should be created as a utility method.

File: pyconpune/tasks.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import settings
+
+_log = logging.getLogger(__name__)
+
+
+if settings.CELERY_BROKER_URL:
+    broker_url = settings.CELERY_BROKER_URL
+else:
+    broker_url = 'amqp://localhost:5672'
```
Comment (sayanchowdhury) [inline pyconpune/tasks.py:13]: We should not hard-code URL, raise an error if the BROKER URL is not set

File: pyconpune/tasks.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+from celery import Celery
```
Comment (sayanchowdhury) [inline pyconpune/tasks.py:1]: The `pyconpune/tasks.py` file should move into `root`.

File: runworker.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+"""
+    Code taken from https://pagure.io/pagure/blob/master/f/runworker.py
```
Comment (sayanchowdhury) [inline runworker.py:2]: This file needs to be created as a django management command.

File: pyconpune/settings/__init__.py [hunk 7]
```diff
@@ -170,3 +170,6 @@
         # release based on the git info.
         'release': raven.fetch_git_sha(os.path.dirname(os.pardir)),
     }
+
+# Celery config
+CELERY_BROKER_URL = None
```
Comment (sayanchowdhury) [inline pyconpune/settings/__init__.py:175]: Why not set the development URL here itself.

**Review by sayanchowdhury — CHANGES_REQUESTED** [review]

File: pyconpune/tasks.py (hunk 4 above)
Comment (vivekanand1101) [reply pyconpune/tasks.py:13]: That was for development, if anyone is feeling lazy

**Comment by vivekanand1101** [comment]
This. This PR. This is one of the saddest thing that has ever happened to me.

**Pull request closed without merging**